nomic/data_operations.py

```diff
@@ -641,130 +574,194 @@ class AtlasMapTags:
+            tile = self.projection.tile_destination / Path(quad_str)
+            tile_dir = tile.parent
+            if tile_dir.exists():
+                tagged_files = tile_dir.glob('*_tag*')
+                for file in tagged_files:
+                    tag_definition_id = file.name.split(".")[-2]
+                    if tag_definition_id in tag_definition_ids:
+                        try:
+                            file.unlink()
+                        except PermissionError:
+                            print("Permission denied: unable to delete outdated tag file. Skipping")
+                            return
+                        except Exception as e:
+                            print(f"Exception occurred when trying to delete outdated tag file: {e}. Skipping")
+                            return
 
-        Returns:
-            True on success.
+    def add(self, ids: List[str], tags: List[str]):
+        # '''
+        # Adds tags to datapoints.
 
-        '''
-        assert isinstance(ids, list), 'datum_ids must be a list of strings'
-        assert isinstance(tags, list), 'tags must be a list of strings'
+        # Args:
+        #     ids: The datum ids you want to tag
+        #     tags: A list containing the tags you want to apply to these data points.
 
-        colname = json.dumps(
-            {
-                'project_id': self.project.id,
-                'atlas_index_id': self.projection.atlas_index_id,
-                'type': 'datum_id',
-                'tags': tags,
-                'delete_all': delete_all,
-            }
-        )
-        payload_table = pa.table([pa.array(ids, type=pa.string())], [colname])
-        buffer = io.BytesIO()
-        writer = ipc.new_file(buffer, payload_table.schema, options=ipc.IpcWriteOptions(compression='zstd'))
-        writer.write_table(payload_table)
-        writer.close()
-        payload = buffer.getvalue()
-
-        headers = self.project.header.copy()
-        headers['Content-Type'] = 'application/octet-stream'
-        response = requests.post(self.project.atlas_api_path + "/v1/project/tag/delete", headers=headers, data=payload)
-        if response.status_code != 200:
-            raise Exception("Failed to delete tags")
+        # '''
+        raise NotImplementedError("AtlasMapTags.add is not supported.")
+
+    def remove(self, ids: List[str], tags: List[str], delete_all: bool = False) -> bool:
+        # '''
+        # Deletes the specified tags from the given data points.
+
+        # Args:
+        #     ids: The datum_ids to delete tags from.
+        #     tags: The list of tags to delete from the data points. Each tag will be applied to all data points in `ids`.
+        #     delete_all: If true, ignores ids parameter and deletes all specified tags from all data points.
+
+        # Returns:
+        #     True on success.
+
+        # '''
+        raise NotImplementedError("AtlasMapTags.remove is not supported.")
```

Comment:
I thought the reason we are going for NotImplemented is because we may/probably will bring it back at some point. Dont mind changing it to deprecated but I think NotImplemented is better as this only impacts users that have been using Python client to add/remove tags and if they see a DeprecatedWarning and then it comes back later it's _slightly_ more confusing than NotImplemented.